File: 01.剑指offer/24.翻转链表/solution.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if head == None or head.next == None:
            return head

        hnode = ListNode()

        while head != None:
            next = head.next
            head.next = hnode.next
            hnode.next = head
            head = next

        return hnode.next

File: 01.剑指offer/24.翻转链表/solution.cpp

struct ListNode {
  ListNode() {
  }
  int val;
  ListNode *next;
  ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
 public:
  ListNode *reverseList(ListNode *head) {
      if (head == nullptr or head->next == nullptr) {
          return head;
      }

      ListNode *hnode = new ListNode();

      while (head != nullptr) {
          ListNode *next = head->next;
          head->next = hnode->next;
          hnode->next = head;
          head = next;
      }

      return hnode->next;
  }
};

